### PR TITLE
Update featurizers docs

### PR DESCRIPTION
**Pull request by nd-02110114**
## Description

I merged `tokenizers.rst` to `featurizer.rst`. This is because it is difficult for new comers to understand the difference among them when the new comers see them in the sidebar of documents.

In addition to this, I fixed some docstring which is not displayed correctly.

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [x] Documentations (modification for documents)

## Checklist

- [x] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [x] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.22.0**)
  - [x] Run `mypy -p deepchem` and check no errors
  - [x] Run `flake8 <modified file> --count` and check no errors
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: docs/source/api_reference/featurizers.rst [hunk 1]
```diff
@@ -195,66 +263,148 @@ lattice and 3D coordinates that specify a periodic crystal structure.
...
+
+The :code:`dc.feat.Featurizer` class is the abstract parent class for all featurizers.
+
+.. autoclass:: deepchem.feat.Featurizer
   :members:
 
-RawFeaturizer
--------------
+MolecularFeaturizer
+^^^^^^^^^^^^^^^^^^^
```
Comment (rbharath) [inline docs/source/api_reference/featurizers.rst:389]: For each of the abstract featurizers, can we add in a sentence explaining when to use this featurizer? For example:

"""
If you're creating a new featurizer that featurizes molecules, you will want to inherit from the abstract `MolecularFeaturizer` base class.
"""

and so on

**Review by rbharath — COMMENTED** [review]
Thanks for the fixes! I think this makes sense to me. Conceptually, the tokenizers are a type of featurizer (although we haven't managed to make the class inheritance work out cleanly yet). Perhaps we should add an abstract class `TokenizerFeaturizer` here?

Also tagging in @seyonechithrananda. Does this rearrangement of the tokenizer docs make sense to you?

I have a small request for extra docs on the abstract classes but otherwise good on my end

File: docs/source/api_reference/featurizers.rst (hunk 1 above)
Comment (seyonechithrananda) [reply docs/source/api_reference/featurizers.rst:389]: I agree! I think the best practical API design is to have MolecularFeaturizer be the abstract base class rather then creating a new separate module for tokenizers. The only time I can see that being useful is if more varying tokenization algorithms for protein sequences, SMILES, SELFIES, etc, come out and until then having them under 'Featurizers' makes the most sense.

Comment (nd-02110114) [reply docs/source/api_reference/featurizers.rst:389]: > For each of the abstract featurizers, can we add in a sentence explaining when to use this featurizer?

I added!

**Comment by nd-02110114** [comment]
I think this PR is ready to merge! 
If there is no comments, I will merge in a couple of days.

**Review by rbharath — APPROVED** [review]
Looks good! Feel free to merge in once ready

**Pull request merged**